res/controllers/Numark-Party-Mix.scripts.js

```diff
@@ -0,0 +1,396 @@
+NumarkPartyMix.autoLoopSizes = [
+    "4",
+    "8",
+    "16",
+    "32"
+];
+
+// dim all lights when inactive instead of turning them off
+components.Button.prototype.off = 0x01;
+
+components.Button.prototype.shutdown = function() {
+    if (this.midi === undefined || this.midi[0] === undefined || this.midi[1] === undefined) {
+        return;
+    }
+    // LEDs are switched off using node off messages
+    // on shutdown LEDs are not dimmed but switched off
+    midi.sendShortMsg(this.midi[0] - 0x10, this.midi[1], 0x00);
+};
+
+// pad modes control codes
+NumarkPartyMix.PadModeControls = {
+    HOTCUE: 0x00,
+    LOOP: 0x0B,
+    SAMPLER: 0x0E,
+    EFX: 0x18,
+};
+
+NumarkPartyMix.init = function(_id, _debugging) {
+    /// init party led switch off
+    midi.sendShortMsg(0xb0, 0x40, 0x00);//0x60 seems to be max
+    midi.sendShortMsg(0xb0, 0x42, 0x00);
+    midi.sendShortMsg(0xb0, 0x43, 0x00);
+
+    // initialize component containers
+    NumarkPartyMix.deck = new components.ComponentContainer();
+    for (var i = 0; i < 2; i++) {
+        NumarkPartyMix.deck[i] = new NumarkPartyMix.Deck(i + 1);
+    }
+
+    NumarkPartyMix.browse = new NumarkPartyMix.Browse();
+    NumarkPartyMix.gains = new NumarkPartyMix.Gains();
+
+    // 0x00 0x01 0x3F is Numark mfg. ID used in SysEx messages.
+    midi.sendSysexMsg([0xF0, 0x00, 0x01, 0x3F, 0x38, 0x48, 0xF7]);
+};
+
+NumarkPartyMix.shutdown = function() {
+    NumarkPartyMix.deck[0].shutdown();
+    NumarkPartyMix.deck[1].shutdown();
+};
+
+NumarkPartyMix.Deck = function(deckNumber) {
+    components.Deck.call(this, deckNumber);
+
+    var channel = deckNumber - 1;
+    var deck = this;
+    this.scratchModeEnabled = false;
+
+    this.playButton = new components.PlayButton({
+        midi: [0x90 + channel, 0x00],
+    });
+
+    this.cueButton = new components.CueButton({
+        midi: [0x90 + channel, 0x01],
+    });
+
+    this.syncButton = new components.SyncButton({
+        midi: [0x90 + channel, 0x02],
+    });
+
+    this.headphoneButton = new components.Button({
+        midi: [0x90 + channel, 0x1B],
+        key: "pfl",
+        output: function(value) {
+            var note = (value === 0x00 ? 0x80 : 0x90) + channel;
+            midi.sendShortMsg(note, 0x1B, this.outValueScale(value));
+        }
+    });
+
+    this.loadButton = new components.Button({
+        midi: [0x9F, 0x01 + channel],
+        inKey: "LoadSelectedTrack"
+    });
+
+    this.volume = new components.Pot({
+        inKey: "volume"
+    });
+
+    this.gain = new components.Pot({
+        inKey: "pregain"
+    });
+
+    this.treble = new components.Pot({
+        group: "[EqualizerRack1_" + this.currentDeck + "_Effect1]",
+        inKey: "parameter3"
+    });
+
+    this.bass = new components.Pot({
+        group: "[EqualizerRack1_" + this.currentDeck + "_Effect1]",
+        inKey: "parameter1"
+    });
+
+    this.pitch = new components.Pot({
+        inKey: "rate",
+        invert: false
```

Comment:
Yes

Reply:
I'm sorry that I'm drilling down this point. Because there have been several bugfix PRs recently that fixed this issue on other controllers.
Can you tell me what midi values the controller sends when you have the tempo slider at the top and bottom? Do you have  `Preferences > Decks > Down increases Speed` enabled? Does the direction of the tempo slider within mixxx match the direction of your hardware?

Reply:
Right now i can only say that i didn't change any default preferences and that the fader moves like it is represented in the gui. (Downward increase speed)

Reply:
top (-%):
Debug [Controller]: Party Mix MIDI 1: t:77147 ms status 0xB0 (ch 1, opcode 0xB), ctrl 0x09, val 0x7F
bottom (+%):
Debug [Controller]: Party Mix MIDI 1: t:128186 ms status 0xB0 (ch 1, opcode 0xB), ctrl 0x09, val 0x00

Reply:
Mhmm, thats odd, but ok.